src/DenoHTTPWorker.ts

```diff
@@ -0,0 +1,377 @@
+            ],
+          },
+        });
+        if (_options.printOutput) {
+          readline.createInterface({ input: stdout }).on("line", (line) => {
+            console.log("[deno]", line);
+          });
+          readline.createInterface({ input: stderr }).on("line", (line) => {
+            console.error("[deno]", line);
+          });
+        }
+
+        worker = new DenoHTTPWorker(
+          _httpSession,
+          port,
+          _got,
+          process,
+          stdout,
+          stderr
+        );
+        running = true;
+        resolve(worker);
+      });
+    };
+    stdout.on("readable", onReadable);
+  });
+};
+export type { DenoHTTPWorker };
+
+class DenoHTTPWorker {
+  #httpSession: http2.ClientHttp2Session;
+  #got: Got;
+  #denoListeningPort: number;
+  #process: ChildProcess;
+  #stdout: Readable;
+  #stderr: Readable;
+  #terminated: Boolean = false;
+
+  constructor(
+    httpSession: http2.ClientHttp2Session,
+    denoListeningPort: number,
+    got: Got,
+    process: ChildProcess,
+    stdout: Readable,
+    stderr: Readable
+  ) {
+    this.#httpSession = httpSession;
+    this.#denoListeningPort = denoListeningPort;
+    this.#got = got;
+    this.#process = process;
+    this.#stdout = stdout;
+    this.#stderr = stderr;
+  }
+
+  get client(): Got {
+    return this.#got;
+  }
+  terminate(code?: number, signal?: string) {
+    if (this.#terminated) {
+      return;
+    }
+    this.onexit(code || this.#process.exitCode || 0, signal || "");
+    this.#terminated = true;
+    if (this.#process && this.#process.exitCode === null) {
+      // TODO: is this preventing listening on SIGINT for cleanup? Do we care?
+      forceKill(this.#process.pid!);
+    }
+    this.#httpSession.close();
+  }
+
+  get stdout() {
+    return this.#stdout;
+  }
+
+  get stderr() {
+    return this.#stderr;
+  }
+
+  get denoListeningPort(): number {
+    return this.#denoListeningPort;
+  }
+  /**
+   * Represents an event handler for the "exit" event. That is, a function to be
+   * called when the Deno worker process is terminated.
+   */
+  onexit: (code: number, signal: string) => void = () => {};
+}
+
+function addOption(
+  list: string[],
+  name: string,
+  option: boolean | string[] | undefined
+) {
+  if (option === true) {
+    list.push(`${name}`);
+  } else if (Array.isArray(option)) {
+    let values = option.join(",");
+    list.push(`${name}=${values}`);
+  }
+}
+
+/**
+ * Forcefully kills the process with the given ID.
+ * On Linux/Unix, this means sending the process the SIGKILL signal.
+ * On Windows, this means using the taskkill executable to kill the process.
+ * @param pid The ID of the process to kill.
+ */
+export function forceKill(pid: number) {
+  // TODO: do we need to SIGINT first to make sure we allow the process to do
+  // any cleanup?
+  const isWindows = /^win/.test(process.platform);
+  if (isWindows) {
+    return killWindows(pid);
+  } else {
+    return killUnix(pid);
+  }
+}
```

Comment:
honestly I'm totally fine with dropping Windows support on the basis of if we don't want to test for it we shouldn't maintain it